pkg/rpc/auth_test.go

```diff
@@ -274,6 +274,72 @@ func TestAuthenticateTenant(t *testing.T) {
 	}
 }
 
+func BenchmarkAuthenticate(b *testing.B) {
+	correctOU := []string{security.TenantsOU}
+	stid := roachpb.SystemTenantID
+	for _, tc := range []struct {
+		name         string
+		systemID     roachpb.TenantID
+		ous          []string
+		commonName   string
+		rootDNString string
+		nodeDNString string
+	}{
+		{name: "tenTen", systemID: stid, ous: correctOU, commonName: "10"},
```

Comment:
Which of the benchmarks reflects the case we're seeing in the sysbench reference workload? Could you add inline comments about when we expect to hit which?